purchase_order_secondary_unit/tests/test_purchase_order_secondary_unit.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2018 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import fields
+from odoo.tests import SavepointCase
+
+
+class TestPurchaseOrderSecondaryUnit(SavepointCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.product_uom_kg = cls.env.ref('uom.product_uom_kgm')
+        cls.product_uom_gram = cls.env.ref('uom.product_uom_gram')
+        cls.product_uom_unit = cls.env.ref('uom.product_uom_unit')
+        cls.product = cls.env['product.product'].create({
+            'name': 'test',
+            'uom_id': cls.product_uom_kg.id,
+            'uom_po_id': cls.product_uom_kg.id,
+            'secondary_uom_ids': [
+                (0, 0, {
+                    'name': 'unit-700',
+                    'uom_id': cls.product_uom_unit.id,
+                    'factor': 0.7,
+                })],
+        })
+        cls.secondary_unit = cls.env['product.secondary.unit'].search([
+            ('product_tmpl_id', '=', cls.product.product_tmpl_id.id),
+        ])
+        cls.product.purchase_secondary_uom_id = cls.secondary_unit.id
+        cls.partner = cls.env['res.partner'].create({
+            'name': 'test - partner',
+            'supplier': True,
+        })
+        type_obj = cls.env['stock.picking.type']
```

Comment:
You do not need create and assign picking_type_id, purchase_stock has a default value for this.
https://github.com/odoo/odoo/blob/12.0/addons/purchase_stock/models/purchase.py#L29
https://github.com/odoo/odoo/blob/12.0/addons/purchase_stock/models/purchase.py#L16